### PR TITLE
docs(keyring) additional notes regarding keyring behavior

### PR DESCRIPTION
**Pull request by p0pr0ck5**

### CODE REVIEW

File: app/enterprise/1.3-x/db-encryption.md [hunk 1]
```diff
@@ -49,6 +49,13 @@ export KONG_KEYRING_PRIVATE_KEY=/path/to/generated/key.pem
 
 All nodes in the Kong cluster should share the same `keyring_enabled` and `keyring_strategy` configuration values. Not every node needs to be provided the management RSA key pair, as that key pair is only used for backup and recovery processes. It does not need to be present for regular database read/write operations.
 
+Note that the user under which Kong worker processes run (as defined by the `nginx_user` Kong configuration option) must have read access to the public and private keys in order to be able to perform keyring export and import operations. We recommend restricting access to these files as tightly as possible. For example:
```
Comment (h3xar0n) [inline app/enterprise/1.3-x/db-encryption.md]:
```suggestion
The Kong worker processes run under the user you define in Kong's configuration property, `nginx_user`. Note that this user must have read access to the public and private keys in order to be able to perform keyring export and import operations. We recommend restricting access to these files as tightly as possible. For example:
```

File: app/enterprise/1.3-x/db-encryption.md [hunk 2]
```diff
@@ -61,6 +68,8 @@ When encryption is enabled on a Kong node, it checks the status of the cluster k
 
 For all other nodes, the generated key will be automatically distributed within a few seconds.
 
+Note that encryption keys are held _only_ in memory, and will not persist past a restart of the Kong process (e.g., running `kong restart`). Because of this, you must export the keyring following it's initialization, or any sensitive fields written with the keyring will not be recoverable if all Kong nodes in a cluster restart simultaneously. Key material _will_ persist after a soft reload of Kong (e.g., `kong reload`).
```
Comment (h3xar0n) [inline app/enterprise/1.3-x/db-encryption.md]:
```suggestion
Note that encryption keys are held _only_ in memory, and do not persist past a restart of the Kong process (e.g., running `kong restart`). Because of this limitation, you must export the keyring following its initialization. Otherwise, if all Kong nodes in a cluster restart simultaneously, any sensitive fields written with the keyring become unrecoverable. Key material still persists after a soft reload of Kong (i.e., `kong reload`).
```

File: app/enterprise/1.3-x/db-encryption.md [hunk 3]
```diff
@@ -294,6 +303,14 @@ Through the kernel CSPRNG, Kong derives keyring material generated through the `
 
 When operating in `cluster` mode, keyring material propagates automatically among all nodes in the Kong cluster. Because Kong nodes do not have a notion of direct peer-to-peer communication, the underlying datastore (Cassandra or PostgreSQL) serves as a communication channel to transmit messages. When a Kong node starts, it generates an ephemeral RSA key pair. The node's public keys propagate to all other active nodes in the cluster. When an active node sees a message request for keyring material, it wraps the in-memory keyring material in the presented public key, and transmits the payload back over the central messaging channel provided by the underlying data store. This process allows each node in the cluster to broadcast keyring material to new nodes, without sending key material in plaintext over the wire. This model requires that at least one node be running at all times within the cluster; a failure of all nodes requires manually re-importing the keyring to one node during an outage recovery.
 
+### Encrypted Fields
+
+The following fields are encrypted as rest by the keyring module:
```
Comment (h3xar0n) [inline app/enterprise/1.3-x/db-encryption.md]:
```suggestion
The keyring module encrypts the following fields at rest:
```

File: app/enterprise/changelog.md [hunk 4]
```diff
@@ -27,6 +27,8 @@ layout: changelog
 
 * **OpenAPI2Kong**: 
     - Adds trace to errors to identify the originating YAML/JSON element that caused the error.
+* **Key Authentication - Encrypted**:
+    - Provides key authentication for Routes and Services, with authentication tokens stored in an encrypted format via encryption-at-rest.
```
Comment (h3xar0n) [inline app/enterprise/changelog.md]:
```suggestion
    - Provides key authentication for Routes and Services, with authentication tokens stored in a format that is encrypted at rest.
```

File: app/enterprise/1.3-x/db-encryption.md (hunk 1 above)
Comment (p0pr0ck5) [reply app/enterprise/1.3-x/db-encryption.md]: to me this reads oddly that we start the sentence discussing the kong user, with no context, followed by a hard stop. this is one thought- "the user must have read access..." vs "there is a user**.** it must have read access..."

will defer to you here

Comment (h3xar0n) [reply app/enterprise/1.3-x/db-encryption.md]: I agree that it should have some context. I wanted to break apart the sentence since I found it hard to follow. Do you think this conveys enough detail?

```suggestion
Note that the user (defined by the `nginx_user` property) must have read access to the public and private keys in order to be able to perform keyring export and import operations. We recommend restricting access to these files as tightly as possible. For example:
```

Comment (p0pr0ck5) [inline app/enterprise/1.3-x/db-encryption.md]:
```suggestion
Note that the user under which Kong worker processes run must have read access to the public and private keys in order to be able to perform keyring export and import operations. This user is defined by the `nginx_user` Kong configuration option. We recommend restricting access to these files as tightly as possible. For example:
```

Comment (p0pr0ck5) [reply app/enterprise/1.3-x/db-encryption.md]: @h3xar0n alternate suggestion? ^

Comment (h3xar0n) [reply app/enterprise/1.3-x/db-encryption.md]: :shipit:
